platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -98,63 +106,50 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(HttpSolrClientFactory.class);
 
   @Override
-  public org.codice.solr.client.solrj.SolrClient newClient(String core) {
-    String solrUrl =
-        StringUtils.defaultIfBlank(
-            AccessController.doPrivileged(
-                (PrivilegedAction<String>) () -> System.getProperty(SOLR_HTTP_URL)),
-            getDefaultHttpsAddress());
-    final String coreUrl = solrUrl + "/" + core;
-    final String solrDataDir =
-        AccessController.doPrivileged(
-            (PrivilegedAction<String>) () -> System.getProperty(SOLR_DATA_DIR));
-
-    if (solrDataDir != null) {
-      ConfigurationStore.getInstance().setDataDirectoryPath(solrDataDir);
+  public org.codice.solr.client.solrj.SolrClient newClient(String coreName) {
+    Args.notEmpty(coreName, "Cannot create Solr client. Missing core name.");
+    if (getSolrDir() != null) {
+      ConfigurationStore.getInstance().setDataDirectoryPath(getSolrDir());
     }
-    LOGGER.debug("Solr({}): Creating an HTTP Solr client using url [{}]", core, coreUrl);
-    return new SolrClientAdapter(core, () -> createSolrHttpClient(solrUrl, core, coreUrl));
+    LOGGER.debug(
+        "Solr({}): Creating an HTTP Solr client using url [{}]", coreName, getCoreUrl(coreName));
+    return new SolrClientAdapter(coreName, () -> createSolrHttpClient(coreName));
   }
 
   @VisibleForTesting
-  SolrClient createSolrHttpClient(String url, String coreName, String coreUrl)
-      throws IOException, SolrServerException {
-    final HttpClientBuilder builder =
-        HttpClients.custom()
-            .setDefaultCookieStore(new BasicCookieStore())
-            .setMaxConnTotal(128)
-            .setMaxConnPerRoute(32);
+  SolrClient createSolrHttpClient(String coreName) throws IOException, SolrServerException {
+    final HttpClientBuilder httpClientBuilder = createHttpBuilder();
+    final HttpSolrClient.Builder solrClientBuilder =
```

Comment:
❓ I do not know why originally the builder was set with 128 max connection total and 32 max per route. Is this no longer required?

Reply:
I don't know why either. The values are still there around line 300. They just got moved into the `createHttpBuilder` method.

```
.setMaxConnTotal(128)
.setMaxConnPerRoute(32)
```

Reply:
This was based on values used by Solr 4.x.
https://github.com/apache/lucene-solr/blob/releases/lucene-solr/4.10.4/solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrServer.java#L165-L166

It looks like the latest Solr code might have significantly increased those limits.
https://github.com/apache/lucene-solr/blob/606e529f9376f3b36d4e3248776fb413f111e03b/solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpClientUtil.java#L80-L81

Either way, I doubt we want the default values due to how that will impact the number of concurrent clients we can support.